### DISCUSSION TITLE
How do I use third party dependencies/libraries in UNO Platform?

### DISCUSSION THREAD

**Discussion by jpgarza93** [unoplatform/uno#10205] [Q&A]
If I want to use for example "Json.NET" must I install the NuGet package in all the platform heads individually?

Are we able to create a cross-targeted library that includes all dependencies?

**Comment by jpgarza93** [comment]
While the information [here ](https://platform.uno/docs/articles/uno-app-solution-structure.html?q=shared%20library) is extremely useful.

I was still not able to tell if I must install it for all projects individually (one at a time) or just one project (like a head project?).

**Reply by jeromelaban** [reply]
This [specific portion](https://platform.uno/docs/articles/uno-app-solution-structure.html?q=shared%20library#handling-dependencies) is about that topic. Head projects (projects creating deployable applications) need to each have dependencies installed.

You can create a library that includes all of those dependencies but know that Visual Studio currently builds all the `TargetFrameworks` of these library projects (if there are more than one). This can take more time to build, and this is why Uno is built around shared projects at this time.
